### ISSUE TITLE
Make release of this provider

### ISSUE THREAD

**Issue by tgermain** [mrolla/terraform-provider-circleci#7]
Currently installing this module can only be done by compiling it from source and copying the binary to the right path.

It would be nice to have properly tagged release with pre-compiled binary on github. 

This can be done with CircleCI as advocate [here](https://circleci.com/blog/publishing-to-github-releases-via-circleci/), but I've never done it myself and I do not have write access to the repository.

To me this release process would act like this : 
1- tag a release following semver versionning
2- This tag triggers a build on CircleCI which build at least linux_amd64 binary (that's what I need for now but it should be easy to add other arch or OS)
3- Publish those binary as part of a github release

**Comment by mrolla** [comment]
I completely agree with you here. CircleCI is already running the (few) tests so it's only a matter of having it cross compile for the oses and push the artifacts back to GH. Will get it done later tonight!
